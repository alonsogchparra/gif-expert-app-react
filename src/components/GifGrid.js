import PropTypes from "prop-types";
import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";

const GifGrid = ({ category }) => {
  const { data: gifs, loading } = useFetchGifs(category);

  return (
    <>
      <hr />
      <div className="col-12 text-center mb-4">
        <h3 className="animate__animated animate__fadeIn">{category}</h3>
      </div>

      {loading && <p>Loading...</p>}

      <div className="row row-cols-1 row-cols-md-3">
        {gifs.map((gif) => (
          <GifGridItem key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
