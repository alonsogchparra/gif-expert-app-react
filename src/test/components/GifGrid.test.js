import { shallow } from "enzyme";
import React from "react";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Test on GifGrid", () => {
  const category = "Breaking Bad";

  test("should show GifGrid correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show items when the images are loaded by useFetchGifs", () => {
    const gifs = [
      {
        id: "Abc",
        url: "https://locahosta/anything.jpg",
        title: "Anything",
      },
      {
        id: "Abc22",
        url: "https://locahosta/anything.jpg",
        title: "Anything",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper.find("GifGridItem").exists()).toBe(true);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
