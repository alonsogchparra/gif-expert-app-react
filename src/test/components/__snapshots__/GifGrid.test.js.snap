// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifGrid should show GifGrid correctly 1`] = `
<Fragment>
  <hr />
  <div
    className="col-12 text-center mb-4"
  >
    <h3
      className="animate__animated animate__fadeIn"
    >
      Breaking Bad
    </h3>
  </div>
  <div
    className="col"
  >
    <div
      className="text-center"
    >
      <div
        class="spinner-border"
        role="status"
      >
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
  <div
    className="row row-cols-1 row-cols-md-3"
  />
</Fragment>
`;
